eng/common/scripts/Helpers/Resource-Helpers.ps1

```diff
@@ -134,3 +149,40 @@ filter Remove-PurgeableResources {
 function Log($Message) {
   Write-Host ('{0} - {1}' -f [DateTime]::Now.ToLongTimeString(), $Message)
 }
+
+function Wait-PurgeableResource {
+  param (
+    [Parameter(Mandatory=$true)]
+    [scriptblock] $ScriptBlock,
+
+    [Parameter(Mandatory=$true)]
+    $Resource,
+
+    [Parameter()]
+    [ValidateRange(1, [int]::MaxValue)]
+    [int] $Timeout = 30,
+
+    [Parameter()]
+    [switch] $PassThru
+  )
+
+  # We build a new AST to pass the `$Resource` as `$r`, which is the `foreach` variable declared in `Remove-PurgeableResources` above.
+  # This is done to make writing a purge script feel natural without having to worry about scope since `[ScriptBlock].GetNewClosure()`
+  # does not capture `$r` appropriately. If the variable name in the `foreach` above is changed, it must be changed here as well.
+  $scriptBlockAst = [System.Management.Automation.Language.Parser]::ParseInput(@"
+  param (`$r)
+  $($ScriptBlock.Ast.EndBlock.ToString())
+"@, [ref] $null, [ref] $null)
+
+  $j = Start-ThreadJob -ScriptBlock $scriptBlockAst.GetScriptBlock() -ArgumentList $Resource
+  $null = Wait-Job -Job $j -Timeout $Timeout
```

Comment:
I have a different changing using `$using:` here but don't want to push it yet because it will cancel the 6+ hour job deleting the remaining KVs from the Test Resources sub. Still more subs to go, but may cancel it after that to test those changes. I ran them locally and it seems to work fine.